.github/workflows/pull-request.yaml

```
@@ -0,0 +1,25 @@
+name: Pull Request
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "master"
+
+jobs:
+  go-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '^1.23.0'
```

We could use go version from the go.mod file.

```suggestion
    - uses: actions/setup-go@v5
      with:
        go-version-file: 'go.mod'
```